src/buffer/out/textBuffer.cpp

```diff
@@ -495,7 +630,7 @@ void TextBuffer::_SetWrapOnCurrentRow() noexcept
 // - fSet - True if this row has a wrap. False otherwise.
 //Return Value:
 // - <none>
-void TextBuffer::_AdjustWrapOnCurrentRow(const bool fSet) noexcept
+void TextBuffer::_AdjustWrapOnCurrentRow(const bool fSet)

```

Comment:
gut feeling: the _current row_ should not need to be faulted in, and so should not be exposed to a commit exception. Is that a faulty understanding on my part?

Reply:
It is possible for the current row to fault if you use CUP (or whatever) to go somewhere that hasn't been previously accessed or rendered yet (which is also possible since the renderer is async). We should decide whether we want to crash under OOM. It'd be a valid strategy. Alternatively we can strategically make use of a hypothetical `GetRowByOffsetNoInit` function that returns `nullptr` when faulting. I would personally not prefer it because it makes the code more complex (= "is it safe to use `NoInit` here? etc.), but I'm absolutely open to implement either change.